### PR TITLE
Refactor signing code

### PR DESCRIPTION
**Pull request by supermassive**
Resolves https://github.com/brave/brave-browser/issues/41121

Refactored parts of core/ui responsible of signing transactions/messages:
- Merged `SignTransactionRequest` and `SignAllTransactionsRequest` into `SignSolTransactionsRequest`. This is SOL only. Both dApp `signTransaction` and `signAllTransactions` are still there but internally there is only one implementation.
- Refactored SimulationService to run SOL simulations separately  for `SignSolTransactionsRequest` and `TransactionInfo`. Also instead of sending ids instead of corresponding payloads from ui to core.
- `TxService.GetTransactionMessageToSign` is moved to proxies. Each explicitly specifies what is returned (vs mojo union).
- Added `EthereumSignatureVRS` `EthereumSignatureBytes` `SolanaSignature` `FilecoinSignature` to distinguish signature type in type-safe manner.

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Browser with this PR should behave in the same way as previous versions for these scenarios:
- Adding HW accounts for Ledger/Trezor for ETH, SOL, FIL
- Transaction signing with Ledger for ETH, SOL, FIL
- Message signing with Ledger for ETH, SOL
- Transaction signing with Trezor for ETH
- Message signing with Trezor for ETH

### CODE REVIEW

**Review by StephenHeaps — APPROVED** [review]

**Review by yrliou — APPROVED** [review]
core code LGTM but this needs an issue and a manual test plan filled, thanks.

**Review by goodov — APPROVED** [review]
`sources-gni-reviewers` lgtm

File: android/java/org/chromium/chrome/browser/app/BraveActivity.java [hunk 1]
```diff
@@ -500,18 +500,19 @@ public WalletModel getWalletModel() {
...
+                    if (pending) {
+                        BraveToolbarLayoutImpl layout = getBraveToolbarLayout();
+                        if (layout != null) {
+                            layout.showWalletPanel();
+                        }
 
-                return;
-            }
-            maybeShowPendingTransactions();
-            maybeShowSignTxRequestLayout();
```
Comment (simoarpe) [inline android/java/org/chromium/chrome/browser/app/BraveActivity.java:513]: Maybe I'm not grasping the whole picture here, but what happened to the ETH pending transactions? The method `maybeShowSignTxRequestLayout()` was in charge of processing both (SOL + ETH pending transaction requests).

Comment (supermassive) [reply android/java/org/chromium/chrome/browser/app/BraveActivity.java:513]: `maybeShowSignTxRequestLayout` -> `SIGN_TRANSACTION`
`maybeShowSignAllTxRequestLayout` -> `SIGN_ALL_TRANSACTIONS`
were both in charge of SOL transactions only. 

Now both cases are covered by a single type
`maybeShowSignSolTransactionsRequestLayout` -> `SIGN_SOL_TRANSACTIONS`

Comment (simoarpe) [reply android/java/org/chromium/chrome/browser/app/BraveActivity.java:513]: Okay, then there's something sketchy going on: the method `maybeShowSignSolTransactionsRequestLayout` is called for all transactions (ETH too). ETH pending transactions need it otherwise the panel won't show up.

Here is a quick recording. Precondition: 1 ETH pending transaction. Observe how the breakpoint for the method `maybeShowSignSolTransactionsRequestLayout` is hit and processed anyway.

[Screencast from 13-09-2024 13:53:02.webm](https://github.com/user-attachments/assets/f8cab8a1-064c-445a-95a0-eac7f1fd2d40)

Here is another recording, same ETH pending transaction, but the method `maybeShowSignSolTransactionsRequestLayout` was commented. Result: the wallet panel is never displayed.

[Screencast from 13-09-2024 13:59:18.webm](https://github.com/user-attachments/assets/236036b6-e513-43ae-b4ac-fd2106df4065)

Let's continue on DMs.

File: android/java/org/chromium/chrome/browser/app/BraveActivity.java [hunk 2]
```diff
@@ -528,28 +529,18 @@ private void maybeShowPendingTransactions() {
...
-                        BraveWalletDAppsActivity.ActivityType.SIGN_TRANSACTION);
-                return;
-            }
-            maybeShowSignAllTxRequestLayout();
-        });
-    }
-
-    private void maybeShowSignAllTxRequestLayout() {
+    // TODO(apaymyshev): refactor this to have a better name.
+    private void maybeShowSignSolTransactionsRequestLayout() {
```
Comment (simoarpe) [inline android/java/org/chromium/chrome/browser/app/BraveActivity.java:533]: I'm going to take care of this in my next PR, entangle the chains and and detach the logic that is not related to SOL transactions!

**Review by simoarpe — APPROVED** [review]
👍 Android part looks good!

File: components/brave_wallet_ui/components/extension/connect-hardware-wallet-panel/index.tsx [hunk 3]
```diff
@@ -99,11 +97,16 @@ export const ConnectHardwareWalletPanel = ({
   const request = signMessageData?.at(0)
   const isSigning = request && request.id !== -1
 
-  const { account: messageAccount } = useAccountQuery(request?.accountId)
+  const { account: singMessageAccount } = useAccountQuery(request?.accountId)
```
Comment (onyb) [inline components/brave_wallet_ui/components/extension/connect-hardware-wallet-panel/index.tsx:100]: nit: typo

**Review by onyb — APPROVED** [review]
frontend ++

Nice work on the typing. Please add a test plan covering the common hardware wallet signing scenarios.
